### PR TITLE
MapShed Show Status

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Adds UI elements that communicate that GIS Data is being gathered for MapShed projects. Only the last two commits are relevant to this PR.

## Demo

Waiting for GIS Data:

image

Waiting for Results:

image

Have Results:

image

Error State:

image

## Testing Instructions

Bundle the scripts.

 * Create a new MapShed project. You should see a spinner as GIS Data is gathered. Once the data is gathered, that screen should be replaced by the Result Type tab headers, each with spinners of their own as they wait for respective results.
 * Reload a saved MapShed project. You should not see a spinner, since GIS Data is already gathered.
 * Create a TR-55 project. You should not see a spinner, as no GIS Data needs to be gathered.
 * Select a very large area. GIS Data step should timeout, and you should see the spinner turn into an error sign.

Builds on top of #1326 
Connects #1298 
Connects #1323

### CODE REVIEW

**Comment by rajadain** [comment]
Rebased on `develop`

**Comment by lewfish** [comment]
When going to the model view before the analyze results are ready, the spinner in the analyze tab is not visible.

**Comment by lewfish** [comment]
In the model view, the 'Gathering Data' spinner is visible as expected for the 'Current Conditions' scenario. But after switching to the 'New Scenario' tab, the spinner is not visible.

**Comment by rajadain** [comment]
Multiple POSTs to `/analyze` are a real problem, refactoring that may be too big for this card. Created #1334 to tackle that.

**Comment by rajadain** [comment]
The fixup above should address the lack of spinner when switching between multiple scenarios

**Comment by rajadain** [comment]
Rebased on develop, merged the fixup into the main commit 6644459. Also added 66331fa to de-duplicate GWLF-E runs.

**Comment by hectcastro** [comment]
@azavea-bot rebuild

**Comment by rajadain** [comment]
@azavea-bot rebuild

**Comment by hectcastro** [comment]
@azavea-bot rebuild

**Comment by lewfish** [comment]
When I create a new GWLF-E model, and click the 'New Scenario' tab, I see 'Gathering Data' and then I the spinners on top of the 'Runoff' and 'Quality' tabs go for a second or two, and then the spinners stop and there is nothing for another 5-10 secs before the results appear. This doesn't seem to happen with TR55.

**Comment by rajadain** [comment]
Okay, I've refactored `fetchResultsIfNeeded` to use the same single-promise technique as being used by `fetchGisDataIfNeeded` and I think it is working in all cases now. Can you test one more time?

File: src/mmw/js/src/core/models.js [hunk 1]
```diff
@@ -213,6 +213,21 @@ var TaskModel = Backbone.Model.extend({
     }
 });
 
+var TaskViewModel = Backbone.Model.extend({
```
Comment (lewfish) [inline src/mmw/js/src/core/models.js]: Consider renaming this to `TaskMessageViewModel` to be consistent.

**Comment by lewfish** [comment]
+1, tested

File: src/mmw/js/src/core/models.js (hunk 1 above)
Comment (rajadain) [inline src/mmw/js/src/core/models.js]: Good call

**Comment by rajadain** [comment]
Reorganized commits. Will merge shortly. Thanks for reviewing.

**Pull request merged**